arbor/backends/event_delivery.md

```diff
@@ -114,7 +114,7 @@ For `fvm_multicell` one integration step comprises:
 Towards the end of the integration period, an integration step may have a zero _dt_
 for one or more cells within the group, and this needs to be handled correctly:
 
-*   Generated mechanism `nrn_state()` methods should be numerically correct with
+*   Generated mechanism `advance_state()` methods should be numerically correct with
```

Comment:
The new names (`advance_state`, `compute_currents`, `init`, `apply_events`) are find with me.

Though I would be happy to stick with `deliver_events`.

Reply:
Regarding the `apply_events` name, I had the compiler complain about 'overloaded virtuals' after moving the definition to `concrete_mechanism`. Although that have gone away after various iterations of clean-up.